docs/variants/protectli_vp2410/recovery.md

```diff
@@ -1,7 +1,99 @@
 # Recovery
 
-Following documentation describes the process of recovering hardware from brick
-state with [RTE](../../../transparent-validation/rte/introduction/) and Dasharo
-open-source firmware.
+## Intro
 
-**This page is under construction**
+The following documentation describes the process of recovering hardware from
+the brick state with [RTE](../../transparent-validation/rte/introduction.md) and
+Dasharo open-source firmware.
+
+## Prerequisites
+
+* [Prepared RTE](../../transparent-validation/rte/v1.1.0/quick-start-guide.md)
+* SOIC-8 Pomona clip
+* 6x female-female wire cables
+
+## Connections
+
+To prepare the stand for flashing follow the steps described below:
+
+1. Connect the wire cables to the Pomona clip.
+
+    ![](../../images/protectli_recovery/pomona_clip.png)
+    ![](../../images/protectli_recovery/pomona_clip_with_cables.png)
+
+1. Connect the Pomona clip to the [SPI header](../../transparent-validation/rte/v1.1.0/specification.md)
+   on RTE.
+
+    | SPI header | Pomona clip  |
+    |:----------:|:------------:|
+    | Vcc        | pin 5 (Vcc)  |
+    | GND        | pin 4 (GND)  |
+    | CS         | pin 1 (CS)   |
+    | SCLK       | pin 7 (CLK)  |
+    | MISO       | pin 2 (MISO) |
+    | MOSI       | pin 8 (MOSI) |
+
+    ![](../../images/protectli_recovery/pomona_clip_with_rte.png)
+
+1. Completely remove the motherboard from the platform cover.
+
+    ![](../../images/protectli_recovery/vp2410_disassembly.png)
+    ![](../../images/protectli_recovery/vp2410_location_of_flash_chip.png)
+
+    > If there is little thermal paste on the CPU, apply it before reassembling
+    > the motherboard to the platform cover.
+
+1. Match pin 1(CS) on the Pomona clip with the first pin of the one of flash
+   chip, marked with a small dot engraved on the chip.
+
+    ![](../../images/protectli_recovery/flash_chip.png)
+    ![](../../images/protectli_recovery/pomona_clip_connected_to_flash_chip.png)
+
+## Firmware flashing
+
+To flash firmware follow the steps described below:
+
+1. Login to RTE via `ssh` or `minicom`.
+1. Turn on the platform by connecting the power supply.
+1. Wait at least 5 seconds.
+1. Turn off the platform by using the power button.
+1. Wait at least 3 seconds.
+1. Set the proper state of the SPI by using the following commands on RTE:
+
+    ```bash
+    # set SPI Vcc to 1.8V
+    echo 0 > /sys/class/gpio/gpio405/value
+    # SPI Vcc on
+    echo 1 > /sys/class/gpio/gpio406/value
+    # SPI lines ON
+    echo 1 > /sys/class/gpio/gpio404/value
+    ```
+
+1. Wait at least 2 seconds.
+1. Disconnect the power supply from the platform.
+1. Wait at least 2 seconds.
+1. Flash the platform by using the following command:
+
+    ```bash
+    flashrom -p linux_spi:dev=/dev/spidev1.0,spispeed=16000 -c "MX25L12835F/MX25L12845E/MX25L12865E" -w [path_to_binary]
```

Comment:
VP2410 binaries are 8MB, so it is impossible to have MX25L128 chip which is 128Mbit = 16MB

Reply:
Updated - https://github.com/Dasharo/docs/pull/559/commits/f49b081c7e16bd94874783a096b6b13572298445